### PR TITLE
Add developer tutorials

### PR DESCRIPTION
**Pull request by kenrogers**
## Description

This PR adds the following tutorials:
- End to end full stack tutorial (tutorial.md)
- Deposit with Leather (deposit.md)
- Withdraw with Leather (withdraw.md)

In addition it updates the structure of the documentation and reorganizes some pieces, mainly sBTC DR information.

### CODE REVIEW

**Review by friedger — COMMENTED** [review]
Looks good. Some smaller comments.

Should we clarify the different version more? Or highlight that this doc is for developer release only? See #99 

We are loosing a lot of information about Nakamoto.

File: src/SUMMARY.md [hunk 1]
```diff
@@ -1,51 +1,49 @@
...
 
-# sBTC Project
+# sBTC Releases
 
 - [sBTC Releases](./sbtc-releases.md)
   - [sBTC 0.1](./sbtc-releases/sbtc-dev.md)
-    - [Reference implementation plan](./sbtc-releases/sbtc-dev/sbtc-dev-reference.md)
-    - [Get Started](./sbtc-releases/sbtc-dev/get-started.md)
-    - [Get Started On Devnet](./sbtc-releases/sbtc-dev/get-started-on-devnet.md)
-    - [FAQ](./sbtc-releases/sbtc-dev/faq.md)
```
Comment (friedger) [inline src/SUMMARY.md:42]: We lost the FAQ. There is a PR with more questions.

File: src/clarity-contracts.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+This native integration with the Bitcoin L1 is one of the great parts of Clarity, as we can verify directly from our smart contracts whether or not a Bitcoin transaction was actually mined, all on chain.
+
+```clojure
+(define-read-only (verify-txid-exists-on-burn-chain (txid (buff 32)) (burn-chain-height uint) (merkle-proof (list 14 (buff 32))) (tx-index uint) (block-header (buff 80)))
+    (contract-call? .clarity-bitcoin-mini was-txid-mined burn-chain-height txid block-header { tx-index: tx-index, hashes: merkle-proof})
+)
+```
+
+This takes in a transaction ID, the Bitcoin block height the transaction was in, and a merkle proof. All of this information is passed to the library to verify that the transaction actually occurred.
```
Comment (friedger) [inline src/clarity-contracts.md:193]: Here could go a link to https://start.bitcoinprimer.dev/

File: src/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,1056 @@
...
+
+## Getting Set Up
+
+Note: There are still some bugs being worked out with the local sBTC devenv. There are code snippets provided in this tutorial and in the Lagoon repo for both devnet and testnet.
+
+For this tutorial, we're going to get you set up with a local version of the sBTC DR. Although this does require a bit more setup time, it will pay off by making your development experience significantly faster by not needing to wait for testnet transactions.
+
+If you prefer to use testnet, you should be able to follow along with a few tweaks that we'll cover along the way.
+
+So, before going any further, make sure you have sBTC set up locally by following the [Get Started on DevNet guide](./sbtc-releases/sbtc-dev/get-started-on-devnet.md).
```
Comment (friedger) [inline src/tutorial.md:57]: by following the "Local environment setup" guide.

File: src/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,1056 @@
...
+        />
+      ) : (
+        ""
+      )}
+    </nav>
+  );
+}
+```
+
+Next we need to create the `ConnectWallet.js` component. You can do that inside the `src/components` directory, which you'll probably also need to create as well.
```
Comment (friedger) [inline src/tutorial.md:195]: the directory was already created for Navbar.js

File: src/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,1056 @@
...
+
+What we have here is a basic UI for converting BTC and sBTC, and borrowing and lending sBTC.
+
+Now that we have our basic UI in place, let's add functionality one piece at a time.
+
+## Initiating a sBTC Deposit
+
+The first thing we are going to do is create a component to initiate a sBTC deposit.
+
+You should already be familiar with how sBTC works at a high level, but what we are going to be doing is constructing a custom Bitcoin transaction that will have all the data we need in order to successfully deposit it into the threshold signature wallet and then mint our sBTC.
```
Comment (friedger) [inline src/tutorial.md:445]: > threshold signature wallet
Is this the official wording? I thought it was "sBTC wallet".

File: src/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,1056 @@
...
+
+Please keep in mind that this is for demo purposes only. Real DeFi lending systems are much more complex and robust than this. Don't use this as a model for building a DeFi protocol, it's only meant to be a starting point for how to work with sBTC.
+
+Now let's write it.
+
+First we're going to set up the data variables we need.
+
+```clojure
+;; Define the contract's data variables
+(define-map deposits { owner: principal } { amount: uint })
```
Comment (friedger) [inline src/tutorial.md:631]: I would not use tuples here

File: src/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,1056 @@
...
+
+Now let's write the function that allows lenders to claim their yield.
+
+```clojure
+(define-public (claim-yield)
+    (let (
+        (user-deposit (default-to u0 (get amount (map-get? deposits { owner: tx-sender }))))
+        (yield-amount (/ (* (var-get pool-reserve) user-deposit) (var-get total-deposits)))
+    )
+        (try! (contract-call? .asset transfer yield-amount (as-contract tx-sender) tx-sender none))
```
Comment (friedger) [inline src/tutorial.md:730]: contract-call needs to be wrapped in `as-contract` and sbtc would be transferred to owner

File: src/SUMMARY.md [hunk 8]
```diff
@@ -1,51 +1,49 @@
...
+# Introduction
 
-[Introduction to sBTC](./introduction.md)
+- [Introduction to sBTC](./introduction.md)
+- [sBTC Roadmap](./sbtc-roadmap.md)
 
 # User Guides
 
 - [How to Deposit and Withdraw](./how-to-deposit-and-withdraw.md)
 - [How to Participate as a Stacker](./how-to-stacker.md)
```
[the remark this answers is not in the document]
Comment (AshtonStephens) [reply src/SUMMARY.md:9]: @kenrogers I think I agree, the roadmap will be more finalized November 10th anyway, and I don't think it's too important that we include it ahead of time.

**Comment by AshtonStephens** [comment]
Are these the same docs that Max is going to be looking through to make his demo app? This looks good on first glance but I'll feel better if someone is thoroughly going through them.

File: src/tutorial.md [hunk 9]
```diff
@@ -0,0 +1,1056 @@
...
+
+    // setting BTC address for testnet
+    const btcAddress = userData.profile.btcAddress.p2wpkh.testnet;
+    const btcPublicKey = userData.profile.btcPublicKey.p2wpkh;
+
+    let utxos = await testnet.fetchUtxos(btcAddress);
+
+    // If we are working via testnet
+    // get sBTC deposit address from bridge API
+    const response = await fetch(
```
Comment (friedger) [inline src/tutorial.md:506]: Remove dependency on sbtc.tech here. The sBTC wallet address should be configurable or a constant for sBTC DR 0.1 
Alternatively, we can read it from the contract directly.

**Comment by kenrogers** [comment]
> Are these the same docs that Max is going to be looking through to make his demo app? This looks good on first glance but I'll feel better if someone is thoroughly going through them.

Yep same docs I sent over to him, he's reviewing them now.

File: src/tutorial.md (hunk 9 above)
Comment (kenrogers) [reply src/tutorial.md:518]: @friedger What is the deployed address of the developer release contract on testnet?

Comment (friedger) [reply src/tutorial.md:518]: https://github.com/stacks-network/sbtc/issues/262#issuecomment-1760140843

**Review by AshtonStephens — APPROVED** [review]

**Pull request merged**